spec/nsqlookup.yml

```diff
@@ -0,0 +1,448 @@
+      description: |
+        Returns a list of all known topics
+      tags:
+        - nsqlookupd
+      responses:
+        200:
+          description: List of topics
+          schema:
+            type: object
+            properties:
+              topics:
+                type: array
+                items:
+                  type: string
+
+  # TODO: Alternate path based on REST
+  # /topics/{topic}/channels:
+  /channels:
+    get:
+      operationId: doChannels
+      summary: Returns a list of all known channels of a topic
+      description: |
+        Returns a list of all known channels of a topic
+      tags:
+        - nsqlookupd
+      parameters:
+        - name: topic
+          in: query
+          description: the topic to list channels for
+          type: string
+          required: true
+      responses:
+        200:
+          description: List of channels
+          schema:
+            type: object
+            properties:
+              channels:
+                type: array
+                items:
+                  type: string
+        400:
+          description: Bad request
+          schema:
+            type: object
+            properties:
+              message:
+                description: Specific error condition
+                type: string
+                default: INVALID_REQUEST
+                enum:
+                  - INVALID_REQUEST
+                  - MISSING_ARG_TOPIC
+        # TODO: Consider TOPIC_ALREADY_EXISTS
+        # 404:
+          # description: Not found
+          # schema:
+            # type: object
+            # properties:
+              # message:
+                # description: Specific error condition
+                # type: string
+                # default: TOPIC_NOT_FOUND
+                # enum:
+                  # - TOPIC_NOT_FOUND
+
+  /nodes:
+    get:
+      operationId: doNodes
+      summary: Returns a list of all known nsqd
+      description: |
+        Returns a list of all known nsqd
+      tags:
+        - nsqlookupd
+      responses:
+        200:
+          description: List of nodes
+          schema:
+            type: object
+            properties:
+              producers:
+                type: array
+                items:
+                  $ref: '#/definitions/Node'
+
+  # TODO: Alternate path based on REST
+  # /topics/{topic}:
+  /topic/create:
+    post:
+      operationId: doCreateTopic
+      summary: Creates a topic
+      description: |
+        Creates a topic
+      tags:
+        - nsqlookupd
+      parameters:
+        - name: topic
+          in: query
+          description: the name of topic to create
+          type: string
+          pattern: '^[\.a-zA-Z0-9_-]+(#ephemeral)?$'
+          minLength: 1
+          maxLength: 64
+          required: true
+      responses:
+        # TODO: Consider 201 or 204
+        200:
+          description: Success
+        # TODO: Consider TOPIC_ALREADY_EXISTS

```

Comment:
Can we consider 409 (Conflict) for TOPIC_ALREADY_EXISTS, so we can ignore the response by just inspecting the HTTP status code instead of parsing the body?